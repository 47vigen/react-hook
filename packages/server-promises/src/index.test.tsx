import React from 'react'
import {renderHook} from '@testing-library/react-hooks'
import {renderToString} from 'react-dom/server'
import {useServerPromises, loadPromises} from './index'

describe('useServerPromises()', () => {
  it('should add promises', () => {
    const {result} = renderHook(() => useServerPromises())
    result.current.push(Promise.resolve('foo'))
    expect(result.current.promises.length).toBe(1)
  })

  it('should render to string w/ resolved promises', async () => {
    const Component = ({value, stop = false}) => {
      const {push} = useServerPromises()

      React.useEffect(() => {
        push(Promise.resolve(value))
      }, [push, value])

      return (
        <div>
          {value} {!stop && <Component value='world' stop />}
        </div>
      )
    }

    const html = await loadPromises(<Component value='Hello' />)
    expect(html).toMatchSnapshot(`
      <div>
        Hello <div>world</div>
      </div>
    `)
  })

  it('should use custom renderer', async () => {
    const Component = ({value, stop = false}) => {
      const {push} = useServerPromises()

      React.useEffect(() => {
        push(Promise.resolve(value))
      }, [push, value])

      return (
        <div>
          {value} {!stop && <Component value='world' stop />}
        </div>
      )
    }

    const html = await loadPromises(<Component value='Hello' />, renderToString)
    expect(html).toMatchSnapshot(`
      <div>
        Hello <div>world</div>
      </div>
    `)
  })
})
