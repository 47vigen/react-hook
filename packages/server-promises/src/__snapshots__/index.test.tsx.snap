// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useServerPromises() should render to string w/ resolved promises: 
      <div>
        Hello <div>world</div>
      </div>
     1`] = `"<div>Hello <div>world </div></div>"`;

exports[`useServerPromises() should use custom renderer: 
      <div>
        Hello <div>world</div>
      </div>
     1`] = `"<div>Hello<!-- --> <div>world<!-- --> </div></div>"`;
